core/test/unit/lib/request_spec.js

```diff
@@ -80,104 +67,103 @@ describe('Request', function () {
 
-    it('[failure] can handle invalid url', function (done) {
-        var url = 'test',
-            options = {
-                headers: {
-                    'User-Agent': 'Mozilla/5.0'
-                }
-            };
-
-        result = request(url, options)
-            .catch(function (err) {
-                should.exist(err);
-                err.message.should.be.equal('URL empty or invalid.');
-                done();
-            });
+    it('[failure] can handle invalid url', function () {
+        const url = 'test';
+        const options = {
+            headers: {
+                'User-Agent': 'Mozilla/5.0'
+            }
+        };
+
+        return request(url, options).then(() => {
+            throw new Error('Request should have rejected with invalid url message');
+        }, (err) => {
+            should.exist(err);
+            err.message.should.be.equal('URL empty or invalid.');
+        });
     });
 
-    it('[failure] can handle empty url', function (done) {
-        var url = '',
-            options = {
-                headers: {
-                    'User-Agent': 'Mozilla/5.0'
-                }
-            };
-
-        result = request(url, options)
-            .catch(function (err) {
-                should.exist(err);
-                err.message.should.be.equal('URL empty or invalid.');
-                done();
-            });
+    it('[failure] can handle empty url', function () {
+        const url = '';
+        const options = {
+            headers: {
+                'User-Agent': 'Mozilla/5.0'
+            }
+        };
+
+        return request(url, options).then(() => {
+            throw new Error('Request should have rejected with invalid url message');
+        }, (err) => {
+            should.exist(err);
+            err.message.should.be.equal('URL empty or invalid.');
+        });
     });
 
-    it('[failure] can handle an error with statuscode not 200', function (done) {
-        var url = 'http://nofilehere.com/files/test.txt',
-            options = {
-                headers: {
-                    'User-Agent': 'Mozilla/5.0'
-                }
-            };
+    it('[failure] can handle an error with statuscode not 200', function () {
+        const url = 'http://nofilehere.com/files/test.txt';
+        const options = {
+            headers: {
+                'User-Agent': 'Mozilla/5.0'
+            }
+        };
 
-        requestMock = nock('http://nofilehere.com')
+        const requestMock = nock('http://nofilehere.com')
             .get('/files/test.txt')
             .reply(404);
 
-        result = request(url, options)
-            .catch(function (err) {
-                requestMock.isDone().should.be.true();
-                should.exist(err);
-                err.statusMessage.should.be.equal('Not Found');
-                done();
-            });
+        return request(url, options).then(() => {
+            throw new Error('Request should have errored');
```

Comment:
:+1: this is a lot more readable and cuts unneeded wtfs when debugging :smiley:

Reply:
Was thinking about it a bit more. Would there be a benefit keeping an error messages for these cases more generic/conventional e.g : `should have errored`? This gives enough context on where it failed and from `.catch` we can easily figure out what exactly was supposed to happen? The upside of making the name generic would be ease of grepping for tests that check error handling :thinking:

Reply:
I had a think about this and i'm on the fence - I'm wondering if it's more useful to be able to search for the ussage of this pattern OR to search for a fairly unique error message... :thinking: